src/main/java/io/confluent/idesidecar/restapi/kafkarest/RecordSerializer.java

```diff
@@ -105,7 +106,20 @@ private ByteString serializeProtobuf(
     try (var protobufSerializer = new KafkaProtobufSerializer<>(client)) {
       protobufSerializer.configure(configs, isKey);
       var schema = (ProtobufSchema) parsedSchema;
-      var record = (Message) wrappedToObject(() -> ProtobufSchemaUtils.toObject(data, schema));
+      var typeRegistry = JsonFormat.TypeRegistry
+          .newBuilder()
+          .add(schema.toDescriptor())
+          .build();
+      var record = (Message) wrappedToObject(() -> {
+        var out = new StringWriter();
+        objectMapper.writeValue(out, data);
+        var message = schema.newMessageBuilder();
+        JsonFormat
+            .parser()
+            .usingTypeRegistry(typeRegistry)
+            .merge(out.toString(), message);
+        return message.build();
```

Comment:
Similarly, we need pass the type registry containing the schema descriptor here so that we're able to properly serialize `Any` values from JSON into Protobuf. Adding `schema.toDescriptor()` also seems to take care of adding dependent (imported) schema descriptors.